### ISSUE TITLE
SOAP Extension - Various Issues

### ISSUE THREAD

**Issue by kingthorin** [zaproxy/zaproxy#4866]
The SOAP extensions and it's scanner rules/plugins are impacted by various issues. 
Per: https://github.com/zaproxy/zap-extensions/pull/1725#issuecomment-407016036
> I noticed other problem (although it might not be the only issue that prevents the scanners from scanning the messages), the message added to `ImportWSDL.requestsList` and put into `ImportWSDL.configurationsList` is later modified (in `WSDLCustomParser.persistMessage`) which means that the scanners are not able to get a `SOAPMsgConfig` nor the actions.

Per https://github.com/zaproxy/zap-extensions/pull/1725#issuecomment-406383321:
> Other things I noticed while tackling this. At https://github.com/zaproxy/zap-extensions/blob/master/addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/ImportWSDL.java#L103 `requestsList` only has entries if the import was just done; if you've opened a saved session or install the addon after the endpoints are already in the Sites Tree then it is empty.

**Software versions**
 - ZAP: 2.7.0 and Dev
 - Add-on: SOAP v3

**Additional context**
https://github.com/zaproxy/zaproxy/issues/4832
https://github.com/zaproxy/zap-extensions/pull/1725

<sub>Testing info: https://github.com/zaproxy/zap-extensions/pull/1725#issuecomment-406381922 </sub>

**Comment by thc202** [comment]
Thanks for raising the issue!

The add-on should not keep the HTTP messages in memory "forever".

**Comment by kingthorin** [comment]
Should we un-publish the add-on (from the marketplace) until this stuff is addressed? With these issues it seems totally in-effective (in my limited testing).

**Comment by thc202** [comment]
The add-on has been removed for the time being.

**Comment by thc202** [comment]
Related #1867.

**Comment by omniaprobate** [comment]
Can the official ZAP documentation be updated with this info?  SOAP Scanner is mentioned as solution number 1 on https://www.zaproxy.org/faq/how-can-you-use-zap-to-scan-apis/ and sounded like exactly what I needed, but when I started up ZAP to look for it in the marketplace I couldn't find it.  It was only then I searched through the Google group and found [this discussion](https://groups.google.com/forum/#!searchin/zaproxy-users/soap$20scanner%7Csort:date/zaproxy-users/MGPXM5dygt4/uyhwc_OuAQAJ) and realized I'd need to just proxy Postman/SoapUI through ZAP instead.
